### DISCUSSION TITLE
Bitcoin wrong signed transaction hex

### DISCUSSION THREAD

**Discussion by apoudineh** [trustwallet/wallet-core#2777] [Q&A]
the problem is (Bitcoin, Doge, ...) UTXO type transaction generated and signed successfully but when i want to broadcast it using blockbook or any other api transaction does not boradcast.
blockbook error: "-25: Missing inputs"
sochain error: "A valid signed transaction hexadecimal string is required. Please check if all inputs in the given transactions are still available to spend. See the \"Is Tx Output Spent?\" API call for reference."

using bitcore-lib-doge generated transaction hex will be broadcasted successfully but trustwallet-core input error

here is trustwallet code: 

            let coin = this.CoinType.doge
            let txInput = TW.Bitcoin.Proto.SigningInput.create({
                hashType: this.BitcoinScript.hashTypeForCoin(coin),
                amount: new Long(198646661),
                toAddress: "DAnpth6ZPZNB3XsZh5mNh9mMPBmpzUasQb",
                changeAddress: "D7CYhSRM56nH7nk5xX2nnbccXUuSznaeGV",
                byteFee: new Long(1),
                privateKey: [this.wallet.getDerivedKey(coin,1,0,1).data()],
                coinType: coin.value,
                utxo: [
                    TW.Bitcoin.Proto.UnspentTransaction.create(
                        {
                            outPoint: {
                                // hash: this.HexCoding.decode("8466181d36fcd7fa9e233e41d89556a1ace30246113b7b62f2acb17452b217df"),
                                hash: this.parseHex("8466181d36fcd7fa9e233e41d89556a1ace30246113b7b62f2acb17452b217df"),
                                index: 0,
                                sequence: 4294967295,
                            },
                            amount: new Long(200000000),
                            script: await this.BitcoinScript.lockScriptForAddress("D7CYhSRM56nH7nk5xX2nnbccXUuSznaeGV", coin).data(),
                        }
                    )
                ],
            })

            const input = TW.Bitcoin.Proto.SigningInput.encode(txInput).finish();
            let txOutput = this.AnySigner.sign(input, coin)
            const output = TW.Bitcoin.Proto.SigningOutput.decode(txOutput);
            const data = JSON.stringify(output.toJSON())
            console.log(this.HexCoding.encode(output.encoded))

bitcore-lib-doge code:
            const sochain_network = "DOGE";
            const recieverAddress = "DAnpth6ZPZNB3XsZh5mNh9mMPBmpzUasQb"

            const privateKey = new PrivateKey('my private key');
            
            const sourceAddress = `D7CYhSRM56nH7nk5xX2nnbccXUuSznaeGV`; 
    
            let fee = 0;
            let inputCount = 0;
            let outputCount = 2;
            const response = {
                "status": "success",
                "data": {
                  "network": "DOGE",
                  "address": "D7CYhSRM56nH7nk5xX2nnbccXUuSznaeGV",
                  "txs": [
              {
                  "txid": "8466181d36fcd7fa9e233e41d89556a1ace30246113b7b62f2acb17452b217df",
                  "output_no": 0,
                  "script_asm": "OP_DUP OP_HASH160 169755d5a6b02014ccd8a1207c1ce98ddeed6c61 OP_EQUALVERIFY OP_CHECKSIG",
                  "script_hex": "76a914169755d5a6b02014ccd8a1207c1ce98ddeed6c6188ac",
                  "value": "2.00000000",
                  "confirmations": 1099,
                  "time": 1670086963
              }
                  ]
                }
            }
            const transaction = new bitcore.Transaction();
            let totalAmountAvailable = 0;
            
             let inputs = [];
             let utxos = response.data.txs;
          
             for (const element of utxos) {
               let utxo = {};
               utxo['satoshis'] = Math.floor(Number(element.value) * 100000000);
               utxo['script'] = element.script_hex;
               utxo['address'] = response.data.address;
               utxo['txId'] = element.txid;
               utxo['outputIndex'] = element.output_no;
               totalAmountAvailable += utxo['satoshis'];
               inputCount += 1;
               inputs.push(utxo);
             }
          
            let transactionSize = inputCount * 146 + outputCount * 34 + 10 - inputCount;

            fee = 1353339

            let satoshiToSend = totalAmountAvailable - fee

            //Set transaction input
            transaction.from(inputs);
          
            // set the recieving address and the amount to send
            transaction.to(recieverAddress, satoshiToSend);
          
            // Set change address - Address to receive the left over funds after transfer
            transaction.change(sourceAddress);
          
             //manually set transaction fees: 20 satoshis per byte
            transaction.fee(fee);
          
            // Sign transaction with your private key
            transaction.sign(privateKey);
          
            const data = JSON.stringify(transaction.toJSON())

            // serialize Transactions
            const serializedTX = transaction.serialize();    
            console.log(serializedTX)

help me with the input problem please

**Comment by optout21** [comment]
A typical issue is that the TXID in the input UTXOs has to be REVERSED.  
See: 

https://github.com/trustwallet/wallet-core/blob/master/src/proto/Bitcoin.proto#L36
https://github.com/trustwallet/wallet-core/blob/master/swift/Tests/Blockchains/BitcoinTests.swift#L91

<img width="736" alt="image" src="https://user-images.githubusercontent.com/13562139/205656851-ff447951-b95a-4c83-add2-1089975bbe45.png">

<img width="877" alt="image" src="https://user-images.githubusercontent.com/13562139/205657004-34dcc372-bc90-49b1-a9e9-9ccb51500583.png">
